allennlp/tests/data/dataset_readers/dataset_reader_test.py

```diff
@@ -1,30 +1,31 @@
 import os
 import shutil
 
+import pytest
 from allennlp.common.testing import AllenNlpTestCase
 from allennlp.data.dataset_readers import SnliReader
 from allennlp.data.dataset_readers.dataset_reader import _LazyInstances
 
 
-class DatasetReaderTest(AllenNlpTestCase):
-    def setUp(self):
-        super().setUp()
-        self.cache_directory = str(self.FIXTURES_ROOT / "data_cache" / "with_prefix")
+class TestDatasetReader:
+    cache_directory = str(AllenNlpTestCase.FIXTURES_ROOT / "data_cache" / "with_prefix")
 
-    def tearDown(self):
-        super().tearDown()
-        shutil.rmtree(self.cache_directory)
+    @pytest.yield_fixture(autouse=True)
+    def cache_directory_fixture(self):
```

Comment:
https://docs.pytest.org/en/latest/yieldfixture.html

FYI, looks like this is deprecated for pytest > 3.0 (I have 4.3.1 installed) and you can just `yield` from a `pytest.fixture()` instead.

Reply:
fa7b32498974f04bf841f3185b6c9fdf34799990